### PR TITLE
Replace WorkloadsOverviewDetailRegistry with reactive solution

### LINKED ISSUE #4816: WorkloadsOverviewDetailRegistry

### PR DESCRIPTION
**Pull request by jansav**
Signed-off-by: Janne Savolainen <janne.savolainen@live.fi>

Closes #4816

### CODE REVIEW

File: src/renderer/components/+workloads-overview/detail-components.injectable.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+const getRegistrationsInPriorityOrder = (
+  allRegistrations: WorkloadsOverviewDetailRegistration[],
+) =>
+  orderBy(
+    "priority",
+    "desc",
+
+    allRegistrations.map(toRegistrationWithDefaultPriority),
+  );
+
```
Comment (Nokel81) [inline src/renderer/components/+workloads-overview/detail-components.injectable.ts:69]: Why is this done differently than done in https://github.com/lensapp/lens/pull/4820?

Comment (jansav) [reply src/renderer/components/+workloads-overview/detail-components.injectable.ts:69]: Use-case  specific logic. This is just extracted to function to make code cleaner.

Comment (Nokel81) [reply src/renderer/components/+workloads-overview/detail-components.injectable.ts:69]: But it seems to be the same logic...

Comment (jansav) [reply src/renderer/components/+workloads-overview/detail-components.injectable.ts:69]: We could share `orderBy("priority", "desc")` if we wanted to, but I don't see reason.

Comment (Nokel81) [reply src/renderer/components/+workloads-overview/detail-components.injectable.ts:69]: See my response on the other PR

Comment (jansav) [reply src/renderer/components/+workloads-overview/detail-components.injectable.ts:69]: Done in #4820

**Review by Iku-turso — DISMISSED** [review]

**Review by aleksfront — DISMISSED** [review]
LGTM

**Review by Iku-turso — APPROVED** [review]

**Review by aleksfront — APPROVED** [review]

**Pull request merged**